bdk_core/src/chain_graph.rs

```diff
@@ -63,8 +63,26 @@ impl<I: ChainIndex> ChainGraph<I> {
         &self,
         update: &Self,
     ) -> Result<ChangeSet<I>, sparse_chain::UpdateFailure<I>> {
+        let mut chain_changeset = self.chain.determine_changeset(&update.chain)?;
+
+        let conflicting_original_txids = update
+            .chain
+            .iter_txids()
+            // choose original txs that conflicts with the update
+            .flat_map(|&(_, txid)| {
+                let update_tx = update.graph.tx(txid).expect("tx should exist");
```

Comment:
I'm not sure we provide the guarantee that this exists? I think that's what #72 was for.